### ISSUE TITLE
Why there is no working carousel for React?

### ISSUE THREAD

**Issue by Maclay74** [brainhubeu/react-carousel#714]
I feel like I tried all of them.
This one was the most promising, but still doesn't work at all.

Why we don't have any working solution? Why for JQuery all of them are working, more or less, but front-end, even more advanced field, has no one single working carousel?

Why am I saying that it doesn't work? Because it doesn't!
When you install it and try to implement the most basic example, you got an error in your console from internal code. 

![image](https://user-images.githubusercontent.com/5504685/120239130-ee424680-c25d-11eb-9ceb-442f36e560d3.png)

Okay, maybe need to go deeper, alright. Maybe it works, but just an error. 

Next problem is that your slides are invisible If you use `collection.map(item => ...)`. By default. Why? The hell knows. 
In order to fix it, need to add settings, plugins. Alright, it fixed too with pain.

Don't you think it's all? Naaaaaah, no way! When you want to render multiple items and one more, like

```
<Carousel>
        {collection.map((item) => (
          <Item key={item.id} />
        ))}
        <OneMoreComponent/>
</Carousel>
```

It won't work! You'll get all your items in one slide and `OneMoreComponent` in another. Why is that happening? 
Okay, maybe render them as fragment? Nah, then we get `Maximum call stack exceeded`.

And this is not only about this carousel. All of them are broken in different way. This one turned out to be top 4-3 I guess.

Why we have plenty solutions for other problems, like, modals, animations and other stuff which just WORK? 
Why there is no single one working carousel for React?

**Comment by Maclay74** [comment]
And, you know, bugs are okay. 
If you open tracker of any widely-used software, there are hundreds of them.

But almost all of them because of advanced usage, when you doing something odd, not trivial. 

I am just trying to follow the README tutorial and faced bug-wall, cmon!

**Comment by Maclay74** [comment]
![image](https://user-images.githubusercontent.com/5504685/120240544-20a17300-c261-11eb-8b44-5e8fb0cf001a.png)

Are you seriously using this? 

Is `React.Children.toArray(this.props.children)` just a joke?
Of course it won't work!

**Comment by Beelphegor** [comment]
> ![image](https://user-images.githubusercontent.com/5504685/120240544-20a17300-c261-11eb-8b44-5e8fb0cf001a.png)
> 
> Are you seriously using this?
> 
> Is `React.Children.toArray(this.props.children)` just a joke?
> Of course it won't work!

Your ramblings are not useful in any way, you should consider contributing to the project instead.

**Comment by shaunstoltz** [comment]
I got this working by used the source and removing all the scss imports as they were bugging in my build environment. Simple import of the index.js file from the src directory here https://github.com/brainhubeu/react-carousel/tree/master/react-carousel/src and the errors about setState were resolved.

**Comment by Maclay74** [comment]
> > ![image](https://user-images.githubusercontent.com/5504685/120240544-20a17300-c261-11eb-8b44-5e8fb0cf001a.png)
> > Are you seriously using this?
> > Is `React.Children.toArray(this.props.children)` just a joke?
> > Of course it won't work!
> 
> Your ramblings are not useful in any way, you should consider contributing to the project instead.

Yes, if this would at least build on local machine. They messed with dependencies and I don't want to spend my day fixing their bugs. 

I'd faster implement it by myself from scratch

**Comment by Maclay74** [comment]
Thanks God, I found it!
Don't waste your time, use `swiper`, it works.
https://swiperjs.com/
